docs/snippets/vue/button-group-story.3.ts.mdx

```diff
@@ -0,0 +1,33 @@
+```ts
+//ButtonGroup.stories.ts
+
+import ButtonGroup from './ButtonGroup.vue';
+
+import { ComponentMeta, ComponentStory } from '@storybook/vue3';
+
+//👇 Imports the Button stories
+import * as ButtonStories from './Button.stories';
+
+export default {
+  /* 👇 The title prop is optional.
+   * See https://storybook.js.org/docs/vue/configure/overview#configure-story-loading
+   * to learn how to generate automatic titles
+   */
+  title: 'ButtonGroup',
+  component: ButtonGroup,
+} as ComponentMeta<typeof Button>;
+
+const Template: ComponentStory<typeof Button> = (args) => ({
+  components: { ButtonGroup },
+  setup() {
+    return { args };
+  },
+  template: '<ButtonGroup v-bind="args" />',
+});
```

Comment:
@valentinpalkovic one thing I've noticed with the snippets is that the types ComponentMeta and ComponentStory don't exist in Vue 2 and Vue 3. I was checking this and adding a story with those types yields the following error:

![vue-3-type-errors](https://user-images.githubusercontent.com/22988955/164813908-6ca53f54-43cc-41a3-a1a9-a5600d751ee8.png)

But both `Meta` and `Story` types do exist and it seems that you could use them as such, for example with Vue 3:

```ts
import Button from "./Button.vue";

import { Meta, Story } from "@storybook/vue3";

// types don't exist in TS for Vue 3
// import { ComponentMeta, ComponentStory } from "@storybook/vue3";

//👇 This default export determines where your story goes in the story list
export default {
  title: "MyButton",
  component: Button,
  argTypes: {
    backgroundColor: { control: "color" },
    onClick: {},
    size: {
      control: { type: "select" },
      options: ["small", "medium", "large"],
    },
  },
} as Meta;

//👇 We create a “template” of how args map to rendering
const Template: Story = (args: unknown) => ({
  components: { Button },
  setup() {
    //👇 The args will now be passed down to the template
    return { args };
  },
  template: '<Button v-bind="args" />',
});
```

And for Vue 2:
```ts
import Button from "./Button.vue";

import { Meta, Story } from "@storybook/vue";

export default {
  title: "Button",
  component: Button,
  argTypes: {
    backgroundColor: { control: "color" },
    size: {
      control: { type: "select" },
      options: ["small", "medium", "large"],
    },
  },
} as Meta;

const Template: Story = (args, { argTypes }) => ({
  props: Object.keys(argTypes),
  components: { Button },
  template: '<Button @onClick="onClick" v-bind="$props" />',
});
```

@shilman are you ok with using this approach with TS+Vue based examples for now? And once and if the types are exported we can adjust the documentation accordingly. Would love to hear your feedback on this.

Reply:
You are totally right. I can't explain, why I have used the wrong types of imports.

@jonniebigodes I have corrected it a bit differently. Instead of using:

```ts
import { Meta, Story } from "@storybook/vue";
import { Meta, Story } from "@storybook/vue3";
```

I have used

```ts
import { Meta, StoryFn } from "@storybook/vue";
import { Meta, StoryFn } from "@storybook/vue3";
```

Because in Storybook v7 `Story` will be the new type for the [Component Story Format 3.0](https://storybook.js.org/blog/component-story-format-3-0/) and StoryFn stays the type for the CSF 2.0 format.

Reply:
@jonniebigodes adjusted!

Reply:
Thanks for taking care of this, appreciate it 🙏 !